test/frontend_test.dart

```diff
@@ -0,0 +1,457 @@
+      });
+
+      tearDown(() {
+        client.close();
+      });
+
+      test('loading site returns valid content', () async {
+        await client.getUrl(Uri.parse(url)).then((HttpClientRequest request) {
+          return request.close();
+        }).then((HttpClientResponse response) {
+          responseCode = response.statusCode;
+        });
+        expect(responseCode == 200, true);
+      });
+    });
+  });
+
+  group("web driver |", () {
+    setUpAll(() async {
+      startSelenium();
+      driver = await setupWebDriver(1280, 1024);
+    });
+
+    tearDownAll(() {
+      stopSelenium();
+      driver.quit();
+      playCompletionBeep();
+    });
+
+    group("basic navigation tests |", () {
+      setUp(() async {
+        await driver.get(url);
+        await waitForAngular(driver);
+      });
+
+      test('base url navigates home', () async {
+        expect(await driver.currentUrl, endsWith('/home'));
+      }, timeout: new Timeout.factor(2));
+
+      test('there is no error messages in the console', () async {
+        bool hasErrors = false;
+        await sleep5();
+        hasErrors = await checkForConsoleErrors(driver, 'SEVERE');
+        expect(await hasErrors, false);
+      });
+
+      test('at least 4 elements in the szolgaltatasok menu', () async {
+        var servicesBtn = await driver.findElement(
+            const By.cssSelector('div.menu-item.menu-item--services > a'));
+        await servicesBtn.click();
+        await sleep2();
+        var menuItemsLength = await driver
+            .findElements(const By.cssSelector(
+                'div.menu-item.menu-item--services > div.nav-menu > div.menu-item'))
+            .length;
+        print("Services menu elements found: " + menuItemsLength.toString());
+        expect(await isXAtLeastValueOfY(menuItemsLength, 4), true);
+      });
+
+      test('szolgaltatasok menu contains an instance of videotar', () async {
+        var servicesBtn = await driver.findElement(
+            const By.cssSelector('div.menu-item.menu-item--services > a'));
+        await servicesBtn.click();
+        var videotarService = await driver.findElement(const By.xpath(
+            "*//div[contains(@class,'menu-item')]/a[contains(text(),'Vide')]"));
+        expect(await elementExists(videotarService), true);
+      });
+
+      test('at least 8 elements under szolgaltatasaink menu', () async {
+        await sleep2();
+        var servicesBtn = await driver.findElement(
+            const By.cssSelector('div.menu-item.menu-item--services > a'));
+        await servicesBtn.click();
+        var menuItemsLength = await driver
+            .findElements(const By.xpath(
+                '*//div[contains(@class, "menu-item--services")]/div[contains('
+                    '@class, "nav-menu")]/div[contains(@class, "menu-item")]'))
+            .length;
+        await waitForAngular(driver);
+        expect(await isXAtLeastValueOfY(menuItemsLength, 8), true);
```

Comment:
Previously there was a test expecting at least 4 menu items under "Szolgáltatások". You are expecting here to have at least 8 elements under the same menu. Is there any difference between the two test (apart from the diff in the expected length of the menu)? I could not spot it.